### PR TITLE
issues/1473 check if model is empty before continuing with remove()

### PR DESCRIPTION
**Pull request by hmottestad**
This PR addresses GitHub issue: #1473 .

Briefly describe the changes proposed in this PR:

* check if model is empty before continuing with remove() method, for performance
* 
*

### CODE REVIEW

File: model/src/main/java/org/eclipse/rdf4j/model/impl/LinkedHashModel.java [hunk 1]
```diff
@@ -167,23 +167,6 @@ public void clear() {
 		statements.clear();
 	}
 
-	@Override
-	public boolean remove(Object o) {
```
Comment (hmottestad) [inline model/src/main/java/org/eclipse/rdf4j/model/impl/LinkedHashModel.java]: removed this because it's implemented in AbstractModel.java.

Comment (jeenbroekstra) [reply model/src/main/java/org/eclipse/rdf4j/model/impl/LinkedHashModel.java]: I notice the implementation is different though. I haven't looked in detail but is the way AbstractModel performs a remove as efficient as the more specific LinkedHashModel implementation?

Comment (hmottestad) [reply model/src/main/java/org/eclipse/rdf4j/model/impl/LinkedHashModel.java]: My understanding is that is just converts the Statements into its parts (subject, predicate...) and calls the other remove(...) method with those parts. 

I will double check though.

Comment (hmottestad) [reply model/src/main/java/org/eclipse/rdf4j/model/impl/LinkedHashModel.java]: Double checked. Essentially identical.

**Pull request merged**